web/src/js/ducks/flows.ts

```diff
@@ -22,17 +24,67 @@ export interface FlowsState extends store.State<Flow> {
     highlight?: string;
     filter?: string;
     sort: { column?: keyof typeof sortFunctions; desc: boolean };
-    selected: string[];
+    selected: Flow[];
+    selectedIndex: { [id: string]: number };
 }
 
 export const defaultState: FlowsState = {
     highlight: undefined,
     filter: undefined,
     sort: { column: undefined, desc: false },
     selected: [],
+    selectedIndex: {},
     ...store.defaultState,
 };
 
+function updateSelected(
+    state: FlowsState = defaultState,
+    newStoreState: State<Flow>,
+    action,
+): Pick<FlowsState, "selected" | "selectedIndex"> {
```

Comment:
We need to fix the lack of test coverage here, otherwise this PR is good to go. 👍

Reply:
I added some tests to cover those lines. Let me know :)